1_prep/src/munge_nmls.R

```diff
@@ -46,4 +51,28 @@ munge_nmls <- function(nml_list_rds, site_ids, base_nml) {
     })
   
   return(nml_objs)
+}
+
+#' @Title Resample incoming hypsography to 1-m intervals
+#' @decription Resample the H and A values to 1-meter intervals; calc bsn_vals
+#' @param H the vector of H (elevation) values from the lake-specific 
+#' nml parameters loaded in from the nml_list.rds from lake-temperature-model-prep
+#' @param A the vector of A (area) values from the lake-specific nml 
+#' parameters loaded in from the nml_list.rds from lake-temperature-model-prep
+#' @return a list containing H and A vectors and a bsn_vals scaler
+resample_H_A <- function(H, A) {
```

Comment:
[I had implemented resampling hypso code in the `lake-temperature-out` pipeline](https://github.com/USGS-R/lake-temperature-out/blob/main/2_process/src/calculate_toha.R#L302-L313) but I did it in a slightly different way. Though, I [am operating on the depth/areas](https://github.com/USGS-R/lake-temperature-out/blob/main/2_process/src/calculate_toha.R#L18-L21), not the H/As.

Reply:
ooh interesting. Thanks for sharing! I'll take a closer look tomorrow